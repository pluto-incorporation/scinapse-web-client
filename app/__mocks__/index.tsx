import { IArticle, recordifyArticle } from "../model/article";
import { IWallet, WalletFactory } from "../model/wallet";
import { IAuthor, recordifyAuthor } from "../model/author";
import { IMember, recordifyMember } from "../model/member";
import { IArticlePoint, ArticlePointFactory } from "../model/articlePoint";
import { IComment, recordifyComment } from "../model/comment";
import { IReviewPoint, ReviewPointFactory } from "../model/reviewPoint";
import { IReview, recordifyReview } from "../model/review";

export const RAW = {
  ARTICLE: require("./article.json") as IArticle,
  ARTICLE_POINT: require("./articlePoint.json") as IArticlePoint,
  AUTHOR: require("./author.json") as IAuthor,
  COMMENT: require("./comment.json") as IComment,
  REVIEW: require("./review.json") as IReview,
  REVIEW_POINT: require("./reviewPoint.json") as IReviewPoint,
  MEMBER: require("./member.json") as IMember,
  WALLET: require("./wallet.json") as IWallet,
};

export const RECORD = {
  ARTICLE: recordifyArticle(RAW.ARTICLE),
  ARTICLE_POINT: ArticlePointFactory(RAW.ARTICLE_POINT),
  AUTHOR: recordifyAuthor(RAW.AUTHOR),
  COMMENT: recordifyComment(RAW.COMMENT),
  REVIEW: recordifyReview(RAW.REVIEW),
  REVIEW_POINT: ReviewPointFactory(RAW.REVIEW_POINT),
  MEMBER: recordifyMember(RAW.MEMBER),
  WALLET: WalletFactory(RAW.WALLET),
};
