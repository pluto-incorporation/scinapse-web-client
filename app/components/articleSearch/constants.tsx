export const LOCAL_STORAGE_FILTERS = "l_s_filters";
export const PREVIOUS_FILTER = "prev_filters";
