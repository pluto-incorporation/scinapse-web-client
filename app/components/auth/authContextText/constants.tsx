export const positiveSignUpContext: { [key: string]: string } = {
  downloadPdf: "📜 To download PDF more, you need to be a Scinapse member.",
  citePaper: "🔍 To copy citation, you need to be a Scinapse member.",
  viewMorePDF: "📜 To view full text, you need to be a Scinapse member.",
  viewMorePDFUnlimited: "📜 To view full text, you need to be a Scinapse member.",
  addToCollection: "🔍 To add papers to collection, you need to be a Scinapse member.",
  clickRequestFullTextBtn: "📜 To request the full text of this paper, you need to be a Scinapse member.",
  query: "🔍 To search more, you need to be a Scinapse member.",
  paperShow: "📈 To view paper information more, you need to be a Scinapse member.",
  paperFromSearch: "📜 To view more papers, you need to be a Scinapse member.",
  queryLover: "💌 Hey Scinapse lover! Please help us to fully support your research by signing up.",
  nextPageFromSearch: "➡ To go to the next page, you need to be a Scinapse member.",
};
