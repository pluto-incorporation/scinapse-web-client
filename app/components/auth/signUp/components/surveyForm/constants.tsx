interface CheckedAnswer {
  name: string;
  order: number;
}

export interface RawQuestionType {
  question: string;
  answers: string[];
  random: boolean;
  type: string;
  description?: string;
}

interface QuestionType {
  question: string;
  random: boolean;
  checked: CheckedAnswer[];
}

export interface RawSurvey {
  surveyName: string;
  questions: RawQuestionType[];
}

export interface Survey {
  surveyName: string;
  questions: QuestionType[];
}
export const SCINAPSE_SURVEY_NAME = 'thinkAsScinapse';

export const SCINAPSE_SURVEY_QUESTIONS: RawSurvey = {
  surveyName: SCINAPSE_SURVEY_NAME,
  questions: [
    {
      question: 'What do you think as Scinapse?',
      answers: ['Publisher', 'PDF Provider', 'Search Engine'],
      random: true,
      type: 'radio',
    },
    { question: 'alpha', answers: ['a', 'b', 'c'], random: true, type: 'checkbox' },
  ],
};
