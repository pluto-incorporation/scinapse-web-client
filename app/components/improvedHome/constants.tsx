export const JOURNALS = [
  'nature',
  'science',
  'ieee',
  'cell',
  'acs',
  'aps',
  'lancet',
  'acm',
  'jama',
  'bmj',
  'pnas',
  'more-journals',
];

export const MOBILE_JOURNALS = ['nature', 'science', 'lancet', 'acm', 'ieee', 'cell', 'more-journal-mobile'];
