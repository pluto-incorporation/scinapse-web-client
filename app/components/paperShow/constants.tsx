export type RELATED_PAPERS = 'reference' | 'cited';
