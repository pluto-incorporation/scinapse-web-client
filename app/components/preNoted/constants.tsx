interface BlockedBubbleKeywords {
  verb: string;
  noun: string;
}

interface BlockedBubbleContextStructure {
  title: string;
  mainText: string;
}

const blockedBubbleVerbAndNoun: { [key: string]: BlockedBubbleKeywords } = {
  downloadPdf: { verb: "download", noun: "PDF" },
  clickRequestFullTextBtn: { verb: "request", noun: "paper" },
  viewMorePDF: { verb: "view", noun: "PDF" },
  addToCollection: { verb: "save to", noun: "Collection" },
};

export function getBlockedBubbleContext(userGroupName: string, buttonAction: string): BlockedBubbleContextStructure {
  const contextVerbAndNoun = blockedBubbleVerbAndNoun[buttonAction];
  switch (userGroupName) {
    case "1":
      return {
        title: `Do you want to ${contextVerbAndNoun.verb} ${contextVerbAndNoun.noun}?`,
        mainText: `Join Scinapse to ${contextVerbAndNoun.verb} ${contextVerbAndNoun.noun}. It's free!`,
      };

    case "2":
      return {
        title: `Join Scinapse to ${contextVerbAndNoun.verb} ${contextVerbAndNoun.noun}`,
        mainText: `When you join Scinapse, you can ${contextVerbAndNoun.verb} any ${contextVerbAndNoun.noun}.`,
      };

    case "3":
      const capitalizedVerb = contextVerbAndNoun.verb.charAt(0).toUpperCase + contextVerbAndNoun.verb.slice(1);
      return {
        title: `${capitalizedVerb} ${contextVerbAndNoun.noun} with a Scinapse account`,
        mainText: `Enjoy unlimited Scinapse with a Scinapse account.`,
      };
  }
  return { title: "", mainText: "" };
}
