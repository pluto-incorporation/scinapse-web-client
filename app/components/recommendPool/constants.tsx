export const BASED_ACTIVITY_COUNT_STORE_KEY = 'recommendPapersDialogCount';
export const ALREADY_VISITED_RECOMMEND_PAPERS = 'visited';
export const BASED_ACTIVITY_PAPER_IDS_FOR_NON_USER_KEY = 'b_a_p_ids';
