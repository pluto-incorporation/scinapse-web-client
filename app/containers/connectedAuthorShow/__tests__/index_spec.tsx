jest.mock("../../../api/author");
jest.mock("react-truncate");

import * as React from "react";
import * as renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MemoryRouter, Route } from "react-router-dom";
import { generateMockStore } from "../../../__tests__/mockStore";
import { initialState } from "../../../reducers";
import ConnectedAuthorShow from "../../../containers/connectedAuthorShow";
import { RAW } from "../../../__mocks__";
import { AUTHOR_SHOW_PATH } from "../../../constants/routes";
import { camelCaseKeys } from "../../../helpers/camelCaseKeys";

describe("ConnectedAuthorShow Component", () => {
  const mappedAuthor = camelCaseKeys(RAW.AUTHOR);
  let mockStore = generateMockStore(initialState);
  let mockState: any;

  describe("when there is no all publications", () => {
    beforeEach(() => {
      mockStore.clearActions();

      mockState = {
        ...initialState,
        connectedAuthorShow: {
          ...initialState.connectedAuthorShow,
          authorId: 2258681040,
          paperIds: [],
          coAuthorIds: [],
          papersTotalPage: 1,
          papersCurrentPage: 1,
          papersTotalCount: 1,
        },
        entities: {
          ...initialState.entities,
          authors: { "2258681040": mappedAuthor },
        },
      };

      mockStore = generateMockStore(mockState);
    });

    it("should render no publications at the content section", () => {
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <MemoryRouter initialIndex={0} initialEntries={["/authors/2258681040"]}>
              <Route path={AUTHOR_SHOW_PATH}>
                <ConnectedAuthorShow />
              </Route>
            </MemoryRouter>
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("when all publications exist", () => {
    beforeEach(() => {
      mockStore.clearActions();

      mockState = {
        ...initialState,
        connectedAuthorShow: {
          ...initialState.connectedAuthorShow,
          authorId: 2258681040,
          paperIds: [RAW.PAPER.id],
          coAuthorIds: [],
          papersTotalPage: 1,
          papersCurrentPage: 1,
          papersTotalCount: 1,
        },
        entities: {
          ...initialState.entities,
          authors: { "2258681040": mappedAuthor },
          papers: { [`${RAW.PAPER.id}`]: RAW.PAPER },
        },
      };

      mockStore = generateMockStore(mockState);
    });

    it("should render content correctly", () => {
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <MemoryRouter initialIndex={0} initialEntries={["/authors/2258681040"]}>
              <Route path={AUTHOR_SHOW_PATH}>
                <ConnectedAuthorShow />
              </Route>
            </MemoryRouter>
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("when try to load all publications", () => {
    beforeEach(() => {
      mockStore.clearActions();

      mockState = {
        ...initialState,
        connectedAuthorShow: {
          ...initialState.connectedAuthorShow,
          authorId: 2258681040,
          isLoadingPapers: true,
          coAuthorIds: [],
          papersTotalPage: 1,
          papersCurrentPage: 1,
          papersTotalCount: 1,
        },
        entities: {
          ...initialState.entities,
          authors: { "2258681040": mappedAuthor },
        },
      };

      mockStore = generateMockStore(mockState);
    });

    it("should render loading spinner correctly", () => {
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <MemoryRouter initialIndex={0} initialEntries={["/authors/2258681040"]}>
              <Route path={AUTHOR_SHOW_PATH}>
                <ConnectedAuthorShow />
              </Route>
            </MemoryRouter>
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
