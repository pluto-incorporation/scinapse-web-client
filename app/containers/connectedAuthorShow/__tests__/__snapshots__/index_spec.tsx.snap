// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedAuthorShow Component when all publications exist should render content correctly 1`] = `
<div
  className="authorShowPageWrapper"
>
  <div
    className="rootWrapper"
  >
    <div
      className="headerBox"
    >
      <div
        className="container"
      >
        <div
          className="leftContentWrapper"
        >
          <div
            className="nameBox"
          >
            <span
              className="nameImgBoxWrapper"
            >
              <div
                className="imgBox"
              >
                H
              </div>
            </span>
            <span
              className="nameHeaderBox"
            >
              <div>
                <span
                  className="username"
                >
                  H. Goldschmidt
                </span>
                 
                <div
                  aria-describedby={null}
                  className="contactIconWrapper"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Verification Author"
                >
                  <i
                    className="icon occupiedIcon"
                  />
                </div>
              </div>
              <div
                className="affiliation"
              >
                Heidelberg University
              </div>
              <div
                className="metricInformation"
              >
                <div
                  className="metricWrapper"
                >
                  <span
                    className="metricValue"
                  >
                    19
                  </span>
                  <span
                    className="metricLabel"
                  >
                    Publications
                  </span>
                </div>
                <div
                  className="metricWrapper"
                >
                  <span
                    className="metricValue"
                  >
                    3,398
                  </span>
                  <span
                    className="metricLabel"
                  >
                    Citations
                  </span>
                </div>
              </div>
              <div
                className="rightBox"
              />
            </span>
          </div>
          <div
            className="mobileMetricInformation"
          >
            <div
              className="metricWrapper"
            >
              <span
                className="metricValue"
              >
                19
              </span>
              <span
                className="metricLabel"
              >
                Publications
              </span>
            </div>
            <div
              className="metricWrapper"
            >
              <span
                className="metricValue"
              >
                3,398
              </span>
              <span
                className="metricLabel"
              >
                Citations
              </span>
            </div>
          </div>
          <div
            className="profileInformationSection"
          >
            <div
              className="bioSection"
            />
            <span
              className="contactSection"
            >
              <a
                className="contactIconWrapper"
                href="mailto:scshinjr@gmail.com"
                rel="noopener"
                target="_blank"
              >
                <i
                  className="icon emailIcon"
                />
              </a>
              <a
                href="mailto:scshinjr@gmail.com"
                rel="noopener"
                target="_blank"
              >
                scshinjr@gmail.com
              </a>
            </span>
          </div>
          <div
            className="tabNavigationWrapper"
          >
            <a
              className="currentTabNavigationItem"
              href="/authors/2258681040"
              onClick={[Function]}
            >
              PUBLICATIONS
            </a>
            <a
              className="tabNavigationItem"
              href="/authors/2258681040/cv"
              onClick={[Function]}
            >
              INFORMATION
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="contentBox"
    >
      <div
        className="container"
      >
        <div
          className="leftContentWrapper"
        >
          <div
            className="allPublicationHeader"
          >
            <span
              className="sectionTitle"
            >
              All Publications
            </span>
            <span
              className="countBadge"
            >
              19
            </span>
            <div
              className="rightBox"
            />
          </div>
          <div
            className="selectedPaperDescription"
          />
          <div
            className="searchSortWrapper"
          >
            <div>
              <div
                className="inputBox"
                style={
                  Object {
                    "backgroundColor": "#f9f9fa",
                    "borderColor": "#f1f3f6",
                    "borderRadius": "4px",
                    "height": "36px",
                    "width": "320px",
                  }
                }
              >
                <input
                  autoFocus={false}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Search paper in author's publication list"
                  value=""
                />
                <div
                  className="icon"
                  onClick={[Function]}
                >
                  <i
                    className="icon"
                  />
                </div>
              </div>
              <div
                className="paperCountMetadata"
              >
                1
                 page of 
                1
                 pages (
                1
                 
                results)
              </div>
            </div>
            <div
              className="rightBox"
            >
              <div
                className="sortBoxWrapper"
              >
                <div
                  className="currentOption"
                  onClick={[Function]}
                >
                  <span
                    className="sortOptionText"
                  >
                    Newest
                  </span>
                  <i
                    className="icon downArrow"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="paperItemWrapper"
          >
            <div
              className="contentSection"
            >
              <div>
                <a
                  className="title"
                  href="/papers/101"
                  onClick={[Function]}
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "Preparation and characterization of graphene oxide paper.",
                      }
                    }
                  />
                </a>
                <a
                  className="newTabIconWrapper"
                  href="/papers/101"
                  onClick={[Function]}
                  rel="noopener"
                  target="_blank"
                >
                  <i
                    className="icon newTabIcon"
                  />
                </a>
              </div>
              <div
                className="publishInfoList"
              >
                <div
                  className="journal"
                >
                  <i
                    className="icon"
                  />
                  <div
                    className="journalText"
                  >
                    <span
                      className="bold"
                    >
                      2007
                       in 
                    </span>
                    <a
                      className="journalName"
                      href="/journals/10"
                      onClick={[Function]}
                    >
                      Nature
                    </a>
                    <span
                      className="bold"
                    >
                       [IF: 40.14]
                    </span>
                  </div>
                </div>
                <div
                  className="author"
                >
                  <i
                    className="icon"
                  />
                  <span
                    className="authors"
                  >
                    <span
                      className="author"
                    >
                      <a
                        className="authorName"
                        href="/authors/undefined"
                        onClick={[Function]}
                      >
                        Dmitriy A. Dikin
                      </a>
                       (Northwestern University)
                      <span>
                        , 
                      </span>
                    </span>
                    <span
                      className="author"
                    >
                      <a
                        className="authorName"
                        href="/authors/undefined"
                        onClick={[Function]}
                      >
                        Sasha Stankovich
                      </a>
                       (Northwestern University)
                      <span>
                        , 
                      </span>
                    </span>
                    <span
                      className="author"
                    >
                      <a
                        className="authorName"
                        href="/authors/undefined"
                        onClick={[Function]}
                      >
                        Eric J. Zimney
                      </a>
                       (Northwestern University)
                    </span>
                    <span
                      className="toggleAuthorsButton"
                      onClick={[Function]}
                    >
                      ... more
                    </span>
                  </span>
                </div>
              </div>
              <div
                className="infoList"
              >
                <a
                  className="referenceButton"
                  href="/papers/101#references"
                  onClick={[Function]}
                >
                  <span>
                    Ref 23
                  </span>
                </a>
                <a
                  className="citedButton"
                  href="/papers/101#cited"
                  onClick={[Function]}
                >
                  <span>
                    Cited 2909
                  </span>
                </a>
                <a
                  className="pdfButton"
                  href="https://www.nature.com/articles/nature06016.pdf"
                  onClick={[Function]}
                  rel="noopener"
                  target="_blank"
                >
                  <i
                    className="icon pdfIconWrapper"
                  />
                  <span>
                    Download PDF
                  </span>
                </a>
                <div
                  className="sourceButton"
                  onClick={[Function]}
                >
                  <i
                    className="icon sourceButtonIcon"
                  />
                  <span>
                    Source
                  </span>
                </div>
                <span
                  className="DOIMetaButtonsWrapper"
                >
                  <span
                    className="citationIconWrapper"
                    onClick={[Function]}
                  >
                    <i
                      className="icon citationIcon"
                    />
                    <span>
                      Cite this paper
                    </span>
                  </span>
                </span>
                <button
                  className="addCollectionBtnWrapper"
                  onClick={[Function]}
                >
                  <i
                    className="icon plusIcon"
                  />
                  <span>
                    Add To Collection
                  </span>
                </button>
                <div
                  className="claimButton"
                >
                  <div>
                    <button
                      className="MuiButtonBase-root-30 MuiIconButton-root-24 additionalMenuIcon"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex="0"
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label-29"
                      >
                        <i
                          className="icon ellipsisIcon"
                        />
                      </span>
                      <span
                        className="MuiTouchRipple-root-33"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="findPaperBtnWrapper"
          >
            <div
              className="findPaperBtn"
              onClick={[Function]}
            >
              Can't find your paper?
            </div>
          </div>
          <div
            className="paginationWrapper"
            style={
              Object {
                "margin": "45px 0 40px 0",
              }
            }
          >
            <div
              className="prevButtons"
            />
            <span
              className="pageItem active"
              onClick={[Function]}
            >
              1
            </span>
          </div>
        </div>
        <div
          className="rightContentWrapper"
        />
      </div>
    </div>
  </div>
  <footer
    className="footerContainer"
  >
    <div
      className="title"
    >
      <div>
        © 2019 Pluto Network. All rights reserved
      </div>
    </div>
    <div
      className="rightBox"
    >
      <a
        className="link"
        href="https://www.facebook.com/PlutoNetwork/"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Facebook
      </a>
      <a
        className="link"
        href="https://twitter.com/pluto_network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Twitter
      </a>
      <a
        className="link"
        href="https://medium.com/pluto-network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Blog
      </a>
      <a
        className="link"
        href="https://pluto.network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        About us
      </a>
      <a
        className="link"
        href="mailto:team@pluto.network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Contact us
      </a>
      <a
        className="link"
        href="https://www.notion.so/pluto/Frequently-Asked-Questions-4b4af58220aa4e00a4dabd998206325c"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        FAQ
      </a>
      <a
        className="termsOfServiceLink"
        href="https://scinapse.io/terms-of-service"
      >
        Terms of service
      </a>
    </div>
  </footer>
</div>
`;

exports[`ConnectedAuthorShow Component when there is no all publications should render no publications at the content section 1`] = `
<div
  className="authorShowPageWrapper"
>
  <div
    className="rootWrapper"
  >
    <div
      className="headerBox"
    >
      <div
        className="container"
      >
        <div
          className="leftContentWrapper"
        >
          <div
            className="nameBox"
          >
            <span
              className="nameImgBoxWrapper"
            >
              <div
                className="imgBox"
              >
                H
              </div>
            </span>
            <span
              className="nameHeaderBox"
            >
              <div>
                <span
                  className="username"
                >
                  H. Goldschmidt
                </span>
                 
                <div
                  aria-describedby={null}
                  className="contactIconWrapper"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Verification Author"
                >
                  <i
                    className="icon occupiedIcon"
                  />
                </div>
              </div>
              <div
                className="affiliation"
              >
                Heidelberg University
              </div>
              <div
                className="metricInformation"
              >
                <div
                  className="metricWrapper"
                >
                  <span
                    className="metricValue"
                  >
                    19
                  </span>
                  <span
                    className="metricLabel"
                  >
                    Publications
                  </span>
                </div>
                <div
                  className="metricWrapper"
                >
                  <span
                    className="metricValue"
                  >
                    3,398
                  </span>
                  <span
                    className="metricLabel"
                  >
                    Citations
                  </span>
                </div>
              </div>
              <div
                className="rightBox"
              />
            </span>
          </div>
          <div
            className="mobileMetricInformation"
          >
            <div
              className="metricWrapper"
            >
              <span
                className="metricValue"
              >
                19
              </span>
              <span
                className="metricLabel"
              >
                Publications
              </span>
            </div>
            <div
              className="metricWrapper"
            >
              <span
                className="metricValue"
              >
                3,398
              </span>
              <span
                className="metricLabel"
              >
                Citations
              </span>
            </div>
          </div>
          <div
            className="profileInformationSection"
          >
            <div
              className="bioSection"
            />
            <span
              className="contactSection"
            >
              <a
                className="contactIconWrapper"
                href="mailto:scshinjr@gmail.com"
                rel="noopener"
                target="_blank"
              >
                <i
                  className="icon emailIcon"
                />
              </a>
              <a
                href="mailto:scshinjr@gmail.com"
                rel="noopener"
                target="_blank"
              >
                scshinjr@gmail.com
              </a>
            </span>
          </div>
          <div
            className="tabNavigationWrapper"
          >
            <a
              className="currentTabNavigationItem"
              href="/authors/2258681040"
              onClick={[Function]}
            >
              PUBLICATIONS
            </a>
            <a
              className="tabNavigationItem"
              href="/authors/2258681040/cv"
              onClick={[Function]}
            >
              INFORMATION
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="contentBox"
    >
      <div
        className="container"
      >
        <div
          className="leftContentWrapper"
        >
          <div
            className="allPublicationHeader"
          >
            <span
              className="sectionTitle"
            >
              All Publications
            </span>
            <span
              className="countBadge"
            >
              19
            </span>
            <div
              className="rightBox"
            />
          </div>
          <div
            className="selectedPaperDescription"
          />
          <div
            className="searchSortWrapper"
          >
            <div>
              <div
                className="inputBox"
                style={
                  Object {
                    "backgroundColor": "#f9f9fa",
                    "borderColor": "#f1f3f6",
                    "borderRadius": "4px",
                    "height": "36px",
                    "width": "320px",
                  }
                }
              >
                <input
                  autoFocus={false}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Search paper in author's publication list"
                  value=""
                />
                <div
                  className="icon"
                  onClick={[Function]}
                >
                  <i
                    className="icon"
                  />
                </div>
              </div>
              <div
                className="paperCountMetadata"
              >
                1
                 page of 
                1
                 pages (
                1
                 
                results)
              </div>
            </div>
            <div
              className="rightBox"
            >
              <div
                className="sortBoxWrapper"
              >
                <div
                  className="currentOption"
                  onClick={[Function]}
                >
                  <span
                    className="sortOptionText"
                  >
                    Newest
                  </span>
                  <i
                    className="icon downArrow"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="noPaperWrapper"
          >
            <div
              className="noPaperDescription"
            >
              There is no publications.
            </div>
          </div>
          <div
            className="findPaperBtnWrapper"
          >
            <div
              className="findPaperBtn"
              onClick={[Function]}
            >
              Can't find your paper?
            </div>
          </div>
          <div
            className="paginationWrapper"
            style={
              Object {
                "margin": "45px 0 40px 0",
              }
            }
          >
            <div
              className="prevButtons"
            />
            <span
              className="pageItem active"
              onClick={[Function]}
            >
              1
            </span>
          </div>
        </div>
        <div
          className="rightContentWrapper"
        />
      </div>
    </div>
  </div>
  <footer
    className="footerContainer"
  >
    <div
      className="title"
    >
      <div>
        © 2019 Pluto Network. All rights reserved
      </div>
    </div>
    <div
      className="rightBox"
    >
      <a
        className="link"
        href="https://www.facebook.com/PlutoNetwork/"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Facebook
      </a>
      <a
        className="link"
        href="https://twitter.com/pluto_network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Twitter
      </a>
      <a
        className="link"
        href="https://medium.com/pluto-network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Blog
      </a>
      <a
        className="link"
        href="https://pluto.network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        About us
      </a>
      <a
        className="link"
        href="mailto:team@pluto.network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Contact us
      </a>
      <a
        className="link"
        href="https://www.notion.so/pluto/Frequently-Asked-Questions-4b4af58220aa4e00a4dabd998206325c"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        FAQ
      </a>
      <a
        className="termsOfServiceLink"
        href="https://scinapse.io/terms-of-service"
      >
        Terms of service
      </a>
    </div>
  </footer>
</div>
`;

exports[`ConnectedAuthorShow Component when try to load all publications should render loading spinner correctly 1`] = `
<div
  className="authorShowPageWrapper"
>
  <div
    className="rootWrapper"
  >
    <div
      className="headerBox"
    >
      <div
        className="container"
      >
        <div
          className="leftContentWrapper"
        >
          <div
            className="nameBox"
          >
            <span
              className="nameImgBoxWrapper"
            >
              <div
                className="imgBox"
              >
                H
              </div>
            </span>
            <span
              className="nameHeaderBox"
            >
              <div>
                <span
                  className="username"
                >
                  H. Goldschmidt
                </span>
                 
                <div
                  aria-describedby={null}
                  className="contactIconWrapper"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Verification Author"
                >
                  <i
                    className="icon occupiedIcon"
                  />
                </div>
              </div>
              <div
                className="affiliation"
              >
                Heidelberg University
              </div>
              <div
                className="metricInformation"
              >
                <div
                  className="metricWrapper"
                >
                  <span
                    className="metricValue"
                  >
                    19
                  </span>
                  <span
                    className="metricLabel"
                  >
                    Publications
                  </span>
                </div>
                <div
                  className="metricWrapper"
                >
                  <span
                    className="metricValue"
                  >
                    3,398
                  </span>
                  <span
                    className="metricLabel"
                  >
                    Citations
                  </span>
                </div>
              </div>
              <div
                className="rightBox"
              />
            </span>
          </div>
          <div
            className="mobileMetricInformation"
          >
            <div
              className="metricWrapper"
            >
              <span
                className="metricValue"
              >
                19
              </span>
              <span
                className="metricLabel"
              >
                Publications
              </span>
            </div>
            <div
              className="metricWrapper"
            >
              <span
                className="metricValue"
              >
                3,398
              </span>
              <span
                className="metricLabel"
              >
                Citations
              </span>
            </div>
          </div>
          <div
            className="profileInformationSection"
          >
            <div
              className="bioSection"
            />
            <span
              className="contactSection"
            >
              <a
                className="contactIconWrapper"
                href="mailto:scshinjr@gmail.com"
                rel="noopener"
                target="_blank"
              >
                <i
                  className="icon emailIcon"
                />
              </a>
              <a
                href="mailto:scshinjr@gmail.com"
                rel="noopener"
                target="_blank"
              >
                scshinjr@gmail.com
              </a>
            </span>
          </div>
          <div
            className="tabNavigationWrapper"
          >
            <a
              className="currentTabNavigationItem"
              href="/authors/2258681040"
              onClick={[Function]}
            >
              PUBLICATIONS
            </a>
            <a
              className="tabNavigationItem"
              href="/authors/2258681040/cv"
              onClick={[Function]}
            >
              INFORMATION
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="contentBox"
    >
      <div
        className="container"
      >
        <div
          className="leftContentWrapper"
        >
          <div
            className="allPublicationHeader"
          >
            <span
              className="sectionTitle"
            >
              All Publications
            </span>
            <span
              className="countBadge"
            >
              19
            </span>
            <div
              className="rightBox"
            />
          </div>
          <div
            className="selectedPaperDescription"
          />
          <div
            className="searchSortWrapper"
          >
            <div>
              <div
                className="inputBox"
                style={
                  Object {
                    "backgroundColor": "#f9f9fa",
                    "borderColor": "#f1f3f6",
                    "borderRadius": "4px",
                    "height": "36px",
                    "width": "320px",
                  }
                }
              >
                <input
                  autoFocus={false}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Search paper in author's publication list"
                  value=""
                />
                <div
                  className="icon"
                  onClick={[Function]}
                >
                  <i
                    className="icon"
                  />
                </div>
              </div>
              <div
                className="paperCountMetadata"
              >
                1
                 page of 
                1
                 pages (
                1
                 
                results)
              </div>
            </div>
            <div
              className="rightBox"
            >
              <div
                className="sortBoxWrapper"
              >
                <div
                  className="currentOption"
                  onClick={[Function]}
                >
                  <span
                    className="sortOptionText"
                  >
                    Newest
                  </span>
                  <i
                    className="icon downArrow"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="spinner"
            style={
              Object {
                "margin": "170px auto",
              }
            }
          >
            <div
              className="bounce1"
            />
            <div
              className="bounce2"
            />
            <div
              className="bounce3"
            />
          </div>
          <div
            className="findPaperBtnWrapper"
          >
            <div
              className="findPaperBtn"
              onClick={[Function]}
            >
              Can't find your paper?
            </div>
          </div>
          <div
            className="paginationWrapper"
            style={
              Object {
                "margin": "45px 0 40px 0",
              }
            }
          >
            <div
              className="prevButtons"
            />
            <span
              className="pageItem active"
              onClick={[Function]}
            >
              1
            </span>
          </div>
        </div>
        <div
          className="rightContentWrapper"
        />
      </div>
    </div>
  </div>
  <footer
    className="footerContainer"
  >
    <div
      className="title"
    >
      <div>
        © 2019 Pluto Network. All rights reserved
      </div>
    </div>
    <div
      className="rightBox"
    >
      <a
        className="link"
        href="https://www.facebook.com/PlutoNetwork/"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Facebook
      </a>
      <a
        className="link"
        href="https://twitter.com/pluto_network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Twitter
      </a>
      <a
        className="link"
        href="https://medium.com/pluto-network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Blog
      </a>
      <a
        className="link"
        href="https://pluto.network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        About us
      </a>
      <a
        className="link"
        href="mailto:team@pluto.network"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        Contact us
      </a>
      <a
        className="link"
        href="https://www.notion.so/pluto/Frequently-Asked-Questions-4b4af58220aa4e00a4dabd998206325c"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        FAQ
      </a>
      <a
        className="termsOfServiceLink"
        href="https://scinapse.io/terms-of-service"
      >
        Terms of service
      </a>
    </div>
  </footer>
</div>
`;
